salt/minion.py

```diff
@@ -3281,20 +3285,14 @@ def destroy(self):
         """
         Tear down the minion
         """
-        if self._running is False:
-            return
```

Comment:
Why did we remove this?

What's the scenario where this is set to false when it shouldn't be?

Reply:
The scenario is that an exception is being raised prior to _running being set but after connections have been opened.